flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/OperatorIdPairList.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobgraph;
+
+import org.apache.flink.runtime.OperatorIdPair;
+
+import java.io.Serializable;
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Convenience class to encapsulate the operator ID pairs of a job vertex. It is convenient because it hides
+ * away the creation of a new list for only operator IDs or user defined operator IDs.
+ * It also hides the iteration over operator ID pairs.
+ */
+public class OperatorIdPairList extends AbstractList<OperatorIdPair> implements Serializable {
```

Comment:
I'm not sure if we need this class:
- `getUserDefinedOperatorIds` is never used
- constructor can be inlined
- instead of `getOperatorIds()` clients could iterate over pairs with minimal changes, e.g. in `StateAssignmentOperation`:
```
List<OperatorIdPair> pairs = executionJobVertex.getOperatorIdPairList().getOperatorIds();

int expectedNumberOfSubTasks = newParallelism * pairs.size();

for (OperatorIdPair pair : operatorIDs) {
  OperatorInstanceID instanceID = OperatorInstanceID.of(subTaskIndex, pair.getOperatorId());
```

Reply:
done